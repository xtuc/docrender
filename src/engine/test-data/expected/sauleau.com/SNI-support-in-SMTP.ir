(
    content: [
        Clear,
        Paragraph([
            InlineText([
                Text("While working on "),
            ], false),
            InlineText([
                TextWithOptions("mailway.app", (
                    link: Some("https://mailway.app/"),
                    styles: [
                        ("text-decoration", "underline"),
                        ("color", "rgb(6.666667%, 33.333336%, 80%)"),
                    ],
                )),
            ], false),
            InlineText([
                Text(" I experimented with routing emails based on "),
            ], false),
            InlineText([
                TextWithOptions("SNI", (
                    link: Some("https://en.wikipedia.org/wiki/Server_Name_Indication"),
                    styles: [
                        ("text-decoration", "underline"),
                        ("color", "rgb(6.666667%, 33.333336%, 80%)"),
                    ],
                )),
            ], false),
            InlineText([
                Text("."),
            ], true),
        ]),
        Clear,
        Paragraph([
            InlineText([
                Text("Let’s imagine a common email setup:"),
            ], true),
        ]),
        Clear,
        Clear,
        List([
            InlineText([
                Text("sauleau.com has a MX record that points to mx.sauleau.com."),
            ], true),
            InlineText([
                Text("mx.sauleau.com has an A record that points to an email server’s IP."),
            ], true),
        ]),
        Clear,
        Paragraph([
            InlineText([
                Text("When someone sends an email to hello@sauleau.com, it needs to resolve the "),
            ], false),
            InlineText([
                TextWithOptions("MX record", (
                    link: Some("https://en.wikipedia.org/wiki/MX_record"),
                    styles: [
                        ("text-decoration", "underline"),
                        ("color", "rgb(6.666667%, 33.333336%, 80%)"),
                    ],
                )),
            ], false),
            InlineText([
                Text(" and the A record to find the corresponding email server."),
            ], true),
        ]),
        Clear,
        Paragraph([
            InlineText([
                Text("Traditionally, mx.sauleau.com would run on its dedicated email server and listen on port 25. To run a secure email server we have a TLS certificate stored on the server issued specifically for mx.sauleau.com."),
            ], true),
        ]),
        Clear,
        Paragraph([
            InlineText([
                Text("The certificate is presented to anyone connecting to the IP and asking for a secure connection. Which implies that it’s not related to the hostname we used to resolve the IP (A record), more importantly, most clients will abort if the TLS certificate doesn’t match the hostname it attempted to connect to."),
            ], true),
        ]),
        Clear,
        Paragraph([
            InlineText([
                Text("To address this issue we could use "),
            ], false),
            InlineText([
                TextWithOptions("SNI", (
                    link: Some("https://en.wikipedia.org/wiki/Server_Name_Indication"),
                    styles: [
                        ("text-decoration", "underline"),
                        ("color", "rgb(6.666667%, 33.333336%, 80%)"),
                    ],
                )),
            ], false),
            InlineText([
                Text(" since it appears earlier in the SMTP communication with the email server and allows to select a TLS certificate to present to the client, which would use the right hostname. However, not all email clients support sending the "),
            ], false),
            InlineText([
                TextWithOptions("SNI", (
                    link: Some("https://en.wikipedia.org/wiki/Server_Name_Indication"),
                    styles: [
                        ("text-decoration", "underline"),
                        ("color", "rgb(6.666667%, 33.333336%, 80%)"),
                    ],
                )),
            ], false),
            InlineText([
                Text("."),
            ], true),
        ]),
        Clear,
        Title((
            level: 2,
            text: Text("Email providers support"),
        ), []),
        Clear,
        Paragraph([]),
        Table((
            rows: [
                (
                    cells: [
                        InlineText([
                            TextWithOptions("Name", (
                                styles: [
                                    ("text-align", "center"),
                                    ("margin-left", "auto"),
                                    ("margin-right", "auto"),
                                ],
                            )),
                        ], false),
                        InlineText([
                            TextWithOptions("Supports SNI?", (
                                styles: [
                                    ("text-align", "center"),
                                    ("margin-left", "auto"),
                                    ("margin-right", "auto"),
                                ],
                            )),
                        ], false),
                    ],
                ),
                (
                    cells: [
                        InlineText([
                            TextWithOptions("Gmail", (
                                link: Some("https://gmail.com/"),
                                styles: [
                                    ("text-decoration", "underline"),
                                    ("color", "rgb(6.666667%, 33.333336%, 80%)"),
                                ],
                            )),
                        ], false),
                        InlineText([
                            Text("yes"),
                        ], false),
                    ],
                ),
                (
                    cells: [
                        InlineText([
                            TextWithOptions("Yahoo", (
                                link: Some("https://mail.yahoo.com/"),
                                styles: [
                                    ("text-decoration", "underline"),
                                    ("color", "rgb(6.666667%, 33.333336%, 80%)"),
                                ],
                            )),
                        ], false),
                        InlineText([
                            Text("no"),
                        ], false),
                    ],
                ),
                (
                    cells: [
                        InlineText([
                            TextWithOptions("AOL", (
                                link: Some("https://mail.aol.com/"),
                                styles: [
                                    ("text-decoration", "underline"),
                                    ("color", "rgb(6.666667%, 33.333336%, 80%)"),
                                ],
                            )),
                        ], false),
                        InlineText([
                            Text("no"),
                        ], false),
                    ],
                ),
                (
                    cells: [
                        InlineText([
                            TextWithOptions("ProtonMail", (
                                link: Some("https://protonmail.com/"),
                                styles: [
                                    ("text-decoration", "underline"),
                                    ("color", "rgb(6.666667%, 33.333336%, 80%)"),
                                ],
                            )),
                        ], false),
                        InlineText([
                            Text("no"),
                        ], false),
                    ],
                ),
                (
                    cells: [
                        InlineText([
                            TextWithOptions("Outlook", (
                                link: Some("https://outlook.live.com/"),
                                styles: [
                                    ("text-decoration", "underline"),
                                    ("color", "rgb(6.666667%, 33.333336%, 80%)"),
                                ],
                            )),
                        ], false),
                        InlineText([
                            Text("yes"),
                        ], false),
                    ],
                ),
                (
                    cells: [
                        InlineText([
                            TextWithOptions("Mailgun", (
                                link: Some("https://www.mailgun.com/"),
                                styles: [
                                    ("text-decoration", "underline"),
                                    ("color", "rgb(6.666667%, 33.333336%, 80%)"),
                                ],
                            )),
                        ], false),
                        InlineText([
                            Text("yes"),
                        ], false),
                    ],
                ),
                (
                    cells: [
                        InlineText([
                            Text("Golang’s "),
                            TextWithOptions("net/smtp", (
                                link: Some("https://golang.org/pkg/net/smtp/"),
                                styles: [
                                    ("text-decoration", "underline"),
                                    ("color", "rgb(6.666667%, 33.333336%, 80%)"),
                                ],
                            )),
                        ], false),
                        InlineText([
                            Text("yes"),
                        ], false),
                    ],
                ),
            ],
        )),
        Clear,
        Paragraph([
            InlineText([
                Text("This list is far from exhaustive, feel free to reach out if you want to try a specific email provider."),
            ], true),
        ]),
        Clear,
    ],
    styles: {
        "pbNORMAL_TEXT": [
            ("line-height", "1.7249999"),
        ],
    },
)