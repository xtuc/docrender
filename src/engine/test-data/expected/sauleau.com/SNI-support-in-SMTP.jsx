import { createElement, Fragment } from "https://unpkg.com/es-react";

function Heading({ style, id, level, children }) {
  const tag = "h" + level;
  return createElement(tag, { style, id }, children);
}

function Paragraph({ children }) {
  return createElement("p", {}, children);
}

function Code({ children }) {
  return createElement("code", {}, children);
}

function Image({ src }) {
  return createElement("img", { src });
}

function Link({ href }) {
  return createElement("a", { href });
}

function List({ children }) {
  return createElement("ul", {}, children);
}

function ListItem({ children }) {
  return createElement("li", {}, children);
}

export default function Doc({
  heading = Heading,
  paragraph = Paragraph,
  code = Code,
  image = Image,
  link = Link,
  list = List,
  listItem = ListItem,
} = {}) {
  return createElement(
    Fragment,
    {},
    createElement(
      paragraph,
      {},
      `While working on `,
      createElement(
        link,
        {
          style: {
            textDecoration: "underline",
            color: "rgb(6.666667%, 33.333336%, 80%)",
          },
          href: "https://mailway.app/",
        },
        `mailway.app`,
      ),
      ` I experimented with routing emails based on `,
      createElement(
        link,
        {
          style: {
            textDecoration: "underline",
            color: "rgb(6.666667%, 33.333336%, 80%)",
          },
          href: "https://en.wikipedia.org/wiki/Server_Name_Indication",
        },
        `SNI`,
      ),
      `.`,
      `
`,
    ),
    createElement(
      paragraph,
      {},
      `Let’s imagine a common email setup:`,
      `
`,
    ),
    createElement(
      list,
      {},
      createElement(
        listItem,
        {},
        `sauleau.com has a MX record that points to mx.sauleau.com.`,
        `
`,
      ),
      createElement(
        listItem,
        {},
        `mx.sauleau.com has an A record that points to an email server’s IP.`,
        `
`,
      ),
    ),
    createElement(
      paragraph,
      {},
      `When someone sends an email to hello@sauleau.com, it needs to resolve the `,
      createElement(
        link,
        {
          style: {
            textDecoration: "underline",
            color: "rgb(6.666667%, 33.333336%, 80%)",
          },
          href: "https://en.wikipedia.org/wiki/MX_record",
        },
        `MX record`,
      ),
      ` and the A record to find the corresponding email server.`,
      `
`,
    ),
    createElement(
      paragraph,
      {},
      `Traditionally, mx.sauleau.com would run on its dedicated email server and listen on port 25. To run a secure email server we have a TLS certificate stored on the server issued specifically for mx.sauleau.com.`,
      `
`,
    ),
    createElement(
      paragraph,
      {},
      `The certificate is presented to anyone connecting to the IP and asking for a secure connection. Which implies that it’s not related to the hostname we used to resolve the IP (A record), more importantly, most clients will abort if the TLS certificate doesn’t match the hostname it attempted to connect to.`,
      `
`,
    ),
    createElement(
      paragraph,
      {},
      `To address this issue we could use `,
      createElement(
        link,
        {
          style: {
            textDecoration: "underline",
            color: "rgb(6.666667%, 33.333336%, 80%)",
          },
          href: "https://en.wikipedia.org/wiki/Server_Name_Indication",
        },
        `SNI`,
      ),
      ` since it appears earlier in the SMTP communication with the email server and allows to select a TLS certificate to present to the client, which would use the right hostname. However, not all email clients support sending the `,
      createElement(
        link,
        {
          style: {
            textDecoration: "underline",
            color: "rgb(6.666667%, 33.333336%, 80%)",
          },
          href: "https://en.wikipedia.org/wiki/Server_Name_Indication",
        },
        `SNI`,
      ),
      `.`,
      `
`,
    ),
    createElement(
      heading,
      { level: 2, id: "pb-todo" },
      `Email providers support`,
    ),
    createElement(paragraph, {}),
    createElement(
      table,
      {},
      createElement(
        tableRow,
        {},
        createElement(
          tableCell,
          {},
          createElement(
            "span",
            {
              style: {
                textAlign: "center",
                marginLeft: "auto",
                marginRight: "auto",
              },
            },
            `Name`,
          ),
        ),
        createElement(
          tableCell,
          {},
          createElement(
            "span",
            {
              style: {
                textAlign: "center",
                marginLeft: "auto",
                marginRight: "auto",
              },
            },
            `Supports SNI?`,
          ),
        ),
      ),
      createElement(
        tableRow,
        {},
        createElement(
          tableCell,
          {},
          createElement(
            link,
            {
              style: {
                textDecoration: "underline",
                color: "rgb(6.666667%, 33.333336%, 80%)",
              },
              href: "https://gmail.com/",
            },
            `Gmail`,
          ),
        ),
        createElement(tableCell, {}, `yes`),
      ),
      createElement(
        tableRow,
        {},
        createElement(
          tableCell,
          {},
          createElement(
            link,
            {
              style: {
                textDecoration: "underline",
                color: "rgb(6.666667%, 33.333336%, 80%)",
              },
              href: "https://mail.yahoo.com/",
            },
            `Yahoo`,
          ),
        ),
        createElement(tableCell, {}, `no`),
      ),
      createElement(
        tableRow,
        {},
        createElement(
          tableCell,
          {},
          createElement(
            link,
            {
              style: {
                textDecoration: "underline",
                color: "rgb(6.666667%, 33.333336%, 80%)",
              },
              href: "https://mail.aol.com/",
            },
            `AOL`,
          ),
        ),
        createElement(tableCell, {}, `no`),
      ),
      createElement(
        tableRow,
        {},
        createElement(
          tableCell,
          {},
          createElement(
            link,
            {
              style: {
                textDecoration: "underline",
                color: "rgb(6.666667%, 33.333336%, 80%)",
              },
              href: "https://protonmail.com/",
            },
            `ProtonMail`,
          ),
        ),
        createElement(tableCell, {}, `no`),
      ),
      createElement(
        tableRow,
        {},
        createElement(
          tableCell,
          {},
          createElement(
            link,
            {
              style: {
                textDecoration: "underline",
                color: "rgb(6.666667%, 33.333336%, 80%)",
              },
              href: "https://outlook.live.com/",
            },
            `Outlook`,
          ),
        ),
        createElement(tableCell, {}, `yes`),
      ),
      createElement(
        tableRow,
        {},
        createElement(
          tableCell,
          {},
          createElement(
            link,
            {
              style: {
                textDecoration: "underline",
                color: "rgb(6.666667%, 33.333336%, 80%)",
              },
              href: "https://www.mailgun.com/",
            },
            `Mailgun`,
          ),
        ),
        createElement(tableCell, {}, `yes`),
      ),
      createElement(
        tableRow,
        {},
        createElement(
          tableCell,
          {},
          `Golang’s `,
          createElement(
            link,
            {
              style: {
                textDecoration: "underline",
                color: "rgb(6.666667%, 33.333336%, 80%)",
              },
              href: "https://golang.org/pkg/net/smtp/",
            },
            `net/smtp`,
          ),
        ),
        createElement(tableCell, {}, `yes`),
      ),
    ),
    createElement(
      paragraph,
      {},
      `This list is far from exhaustive, feel free to reach out if you want to try a specific email provider.`,
      `
`,
    ),
  );
}
