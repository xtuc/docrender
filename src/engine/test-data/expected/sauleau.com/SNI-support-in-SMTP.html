<style>
  .pbNORMAL_TEXT {
    line-height: 1.7249999;
  }
</style>
<p>
  While working on
  <a
    style="text-decoration: underline; color: rgb(6.666667%, 33.333336%, 80%)"
    href="https://mailway.app/"
    >mailway.app</a
  >
  I experimented with routing emails based on
  <a
    style="text-decoration: underline; color: rgb(6.666667%, 33.333336%, 80%)"
    href="https://en.wikipedia.org/wiki/Server_Name_Indication"
    >SNI</a
  >.
</p>
<p>Let’s imagine a common email setup:</p>
<ul>
  <li>sauleau.com has a MX record that points to mx.sauleau.com.</li>
  <li>mx.sauleau.com has an A record that points to an email server’s IP.</li>
</ul>
<p>
  When someone sends an email to hello@sauleau.com, it needs to resolve the
  <a
    style="text-decoration: underline; color: rgb(6.666667%, 33.333336%, 80%)"
    href="https://en.wikipedia.org/wiki/MX_record"
    >MX record</a
  >
  and the A record to find the corresponding email server.
</p>
<p>
  Traditionally, mx.sauleau.com would run on its dedicated email server and
  listen on port 25. To run a secure email server we have a TLS certificate
  stored on the server issued specifically for mx.sauleau.com.
</p>
<p>
  The certificate is presented to anyone connecting to the IP and asking for a
  secure connection. Which implies that it’s not related to the hostname we used
  to resolve the IP (A record), more importantly, most clients will abort if the
  TLS certificate doesn’t match the hostname it attempted to connect to.
</p>
<p>
  To address this issue we could use
  <a
    style="text-decoration: underline; color: rgb(6.666667%, 33.333336%, 80%)"
    href="https://en.wikipedia.org/wiki/Server_Name_Indication"
    >SNI</a
  >
  since it appears earlier in the SMTP communication with the email server and
  allows to select a TLS certificate to present to the client, which would use
  the right hostname. However, not all email clients support sending the
  <a
    style="text-decoration: underline; color: rgb(6.666667%, 33.333336%, 80%)"
    href="https://en.wikipedia.org/wiki/Server_Name_Indication"
    >SNI</a
  >.
</p>
<h2 id="pb-Email%20providers%20support">Email providers support</h2>
<table>
  <tr>
    <td>
      <span style="text-align: center; margin-left: auto; margin-right: auto"
        >Name</span
      >
    </td>
    <td>
      <span style="text-align: center; margin-left: auto; margin-right: auto"
        >Supports SNI?</span
      >
    </td>
  </tr>
  <tr>
    <td>
      <a
        style="
          text-decoration: underline;
          color: rgb(6.666667%, 33.333336%, 80%);
        "
        href="https://gmail.com/"
        >Gmail</a
      >
    </td>
    <td>yes</td>
  </tr>
  <tr>
    <td>
      <a
        style="
          text-decoration: underline;
          color: rgb(6.666667%, 33.333336%, 80%);
        "
        href="https://mail.yahoo.com/"
        >Yahoo</a
      >
    </td>
    <td>no</td>
  </tr>
  <tr>
    <td>
      <a
        style="
          text-decoration: underline;
          color: rgb(6.666667%, 33.333336%, 80%);
        "
        href="https://mail.aol.com/"
        >AOL</a
      >
    </td>
    <td>no</td>
  </tr>
  <tr>
    <td>
      <a
        style="
          text-decoration: underline;
          color: rgb(6.666667%, 33.333336%, 80%);
        "
        href="https://protonmail.com/"
        >ProtonMail</a
      >
    </td>
    <td>no</td>
  </tr>
  <tr>
    <td>
      <a
        style="
          text-decoration: underline;
          color: rgb(6.666667%, 33.333336%, 80%);
        "
        href="https://outlook.live.com/"
        >Outlook</a
      >
    </td>
    <td>yes</td>
  </tr>
  <tr>
    <td>
      <a
        style="
          text-decoration: underline;
          color: rgb(6.666667%, 33.333336%, 80%);
        "
        href="https://www.mailgun.com/"
        >Mailgun</a
      >
    </td>
    <td>yes</td>
  </tr>
  <tr>
    <td>
      Golang’s
      <a
        style="
          text-decoration: underline;
          color: rgb(6.666667%, 33.333336%, 80%);
        "
        href="https://golang.org/pkg/net/smtp/"
        >net/smtp</a
      >
    </td>
    <td>yes</td>
  </tr>
</table>
<p>
  This list is far from exhaustive, feel free to reach out if you want to try a
  specific email provider.
</p>
